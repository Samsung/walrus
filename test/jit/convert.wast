(module
(func (export "test1") (result i32 i64 i64) (local i64 i32)
  i64.const 0x7fffffff80000001
  local.set 0

  i32.const 0x80000001
  local.set 1

  local.get 0
  i32.wrap_i64

  local.get 1
  i64.extend_i32_s

  local.get 1
  i64.extend_i32_u

  (; 2147483649, 18446744071562067969, 2147483649 ;)
)
)

(assert_return (invoke "test1") (i32.const 2147483649) (i64.const 18446744071562067969) (i64.const 2147483649))
