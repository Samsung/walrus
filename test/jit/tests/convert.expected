[[[[[[[  Function   0  ]]]]]]]
Local 0: mapped to: 0 range [1 - 4]
Local 1: mapped to: 1 range [3 - 8]
Number of locals reduced from 2 to 2
0: Opcode: i64.const
  Result[0]: i64
1: Opcode: local.set
  Index: 0
  Param[0]: ref:0(0)
2: Opcode: i32.const
  Result[0]: i32
3: Opcode: local.set
  Index: 1
  Param[0]: ref:2(0)
4: Opcode: local.get
  Index: 0
  Result[0]: i64
5: Opcode: i32.wrap_i64
  Param[0]: ref:4(0)
  Result[0]: i32
6: Opcode: local.get
  Index: 1
  Result[0]: i32
7: Opcode: i64.extend_i32_s
  Param[0]: ref:6(0)
  Result[0]: i64
8: Opcode: local.get
  Index: 1
  Result[0]: i32
9: Opcode: i64.extend_i32_u
  Param[0]: ref:8(0)
  Result[0]: i64
10: Opcode: return
  Param[0]: ref:5(0)
  Param[1]: ref:7(0)
  Param[2]: ref:9(0)
------------------------------
Frame size: 0, Arguments size: 48 (total: 48)
0: Opcode: i64.const
  Result[0]: i64 unused
1: Opcode: local.set
  Addr: sp[24]
  Param[0]: i64 const:0
2: Opcode: i32.const
  Result[0]: i32 unused
3: Opcode: local.set
  Addr: sp[32]
  Param[0]: i32 const:2
4: Opcode: local.get
  Addr: sp[24]
  Result[0]: i64 unused
5: Opcode: i32.wrap_i64
  Param[0]: i64 sp[24]
  Result[0]: i32 sp[0]
6: Opcode: local.get
  Addr: sp[32]
  Result[0]: i32 unused
7: Opcode: i64.extend_i32_s
  Param[0]: i32 sp[32]
  Result[0]: i64 sp[8]
8: Opcode: local.get
  Addr: sp[32]
  Result[0]: i32 unused
9: Opcode: i64.extend_i32_u
  Param[0]: i32 sp[32]
  Result[0]: i64 sp[16]
10: Opcode: return
  Param[0]: i32 sp[0]
  Param[1]: i64 sp[8]
  Param[2]: i64 sp[16]

test1() => i32:2147483649, i64:18446744071562067969, i64:2147483649
